front-end/src/components/CreateEditPostModal.tsx

```diff
@@ -1,23 +1,47 @@
-import axios, { AxiosRequestConfig } from "axios";
+import axios, { AxiosRequestConfig, AxiosResponse } from "axios";
 import React, { useState } from "react";
 import { Alert, Form, FormGroup, Input, Label, Modal, ModalBody, ModalFooter, ModalHeader } from "reactstrap";
+import { preProcessFile } from "typescript";
+import { Post } from "../types/Post";
 import { UserLogin } from "../types/UserLogin";
 
 interface Props {
   loggedInUser: UserLogin
   toggle: any
   isModalOpen: boolean
+  // These are fields of a post that you want to edit. If this is undefined, the modal will create posts.
+  // If defined, modal will edit an existing post.
+  editFields?: Post
+  // Used to append the created post to the feed. Otherwise the feed will be outdated
+  prependToFeed?: Function
 }
 
-export default function CreatePostComponent(props: Props){
+/**
+ * If editFields is not undefined, then this modal will act as an editing modal
+ * @param props 
+ */
+export default function CreateEditPostModal(props: Props){
 
-  const [title, setTitle] = useState('');
-  const [desc, setDesc] = useState('');
-  const [contentType, setContentType] = useState('Text');
-  const [content, setContent] = useState('');
-  const [categories, setCategories] = useState<string[]>([""])
-  const [visibility, setVisibility] = useState('Public');
-  const [unlisted, setUnlisted] = useState(false);
+  const emptyPostFields = {
+    title:'',
+    description:'',
+    contentType:'text/plain',
+    content:'',
+    categories:[''],
+    visibility:'Public',
+    unlisted:false
+  }
+
+  const postFields = props.editFields ?? emptyPostFields
+  const isCreate = props.editFields === undefined
+
+  const [title, setTitle] = useState(postFields.title);
+  const [desc, setDesc] = useState(postFields.description);
+  const [contentType, setContentType] = useState(postFields.contentType);
+  const [content, setContent] = useState(postFields.content);
+  const [categories, setCategories] = useState<string[]>(postFields.categories);
+  const [visibility, setVisibility] = useState(postFields.visibility);
+  const [unlisted, setUnlisted] = useState(postFields.unlisted);
   const [showError, setShowError] = useState(false);
```

Comment:
I think we also need to be able to have author's be able to create a post with a given post_id (in the spec, we should PUT to create a post with an author specified ID. I'm not sure if this will require a change on the backend model.

Reply:
on second thought - we can add author-specified post_id's later. nbd, I'll just open up a new/separate issue for it since I think it can be worked on independently from everything else in this PR/issue

Reply:
Good catch! I agree that a separate issue would be good. Hopefully, it should be small enough for anyone to do if they have some time